File: [2018-07-20_2018-07-25]angular/code/firstangular/src/app/metier/produit.ts

export class Produit {
    public constructor(public id: number,
                       public nom: string,
                       public prix: number) {}

    public afficher() : string {
        return "produit: " + this.nom + ", " + this.prix;
    }

    public description() : string {
        return `produit ${this.nom} de prix ${this.prix}`;
    }
}

File: [2018-08-20_2018-08-28]javaee_spring_jpa/code/secondJpaForm/src/main/java/com/courtalon/secondJpaForm/beans/Produit.java
package com.courtalon.secondJpaForm.beans;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.Table;

import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.ToString;

@Getter @Setter @NoArgsConstructor @ToString
@Entity
@Table(name="products")  // le nom de la table est personnalisable
public class Produit {
	@GeneratedValue(strategy=GenerationType.IDENTITY)
	@Id  						private int id;
	@Column(length=100)			private String nom;
								private double prix;
	@Column(name="WEIGHT")		private double poids;
	
	public Produit(int id, String nom, double prix, double poids) {
		this.id = id;
		this.nom = nom;
		this.prix = prix;
		this.poids = poids;
	}
	

}
